File: Logitude/AngularModules/AngularModules/Common/EntityPMs/DocumentPM.ts
import {UIProperties, UIProperty} from '../../Infrastructure/Components/LogitudeComponents/UIProperties';
export class DocumentPM {

    public UIProperties: UIProperties;
    constructor() {
        this.UIProperties = new UIProperties;
        this.IsDirty = false;
    }


    private id: string;
    public get Id() { return this.id; }
    public set Id(newValue: string) { this.id = newValue; this.MarkAsDirty(); }

    private fileName: string;
    public get FileName() { return this.fileName; }
    public set FileName(newValue: string) { this.fileName = newValue; this.MarkAsDirty(); }


    private fileSize: number;
    public get FileSize() { return this.fileSize; }
    public set FileSize(newValue: number) { this.fileSize = newValue; this.MarkAsDirty(); }

    private extension: string;
    public get Extension() { return this.extension; }
    public set Extension(newValue: string) { this.extension = newValue; this.MarkAsDirty(); }

    private calculatedFileName: string;
    public get CalculatedFileName() { return this.calculatedFileName; }
    public set CalculatedFileName(newValue: string) { this.calculatedFileName = newValue; this.MarkAsDirty(); }
    
    public IsDirty: boolean;
    MarkAsDirty() {
        this.IsDirty = true;

    }

}

File: AmitalCloud/AmitalCloud.Infrastructure.Domain/EntityPMs/Manual/DocumentPM.cs
using AmitalCloud.Infrastructure.Domain.EntityPOCOs;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AmitalCloud.Infrastructure.Domain.EntityPMs
{
    public partial class DocumentPM
    {
        public DocumentPM() : base() { }
        public DocumentPM(Document entity) : base()
        {
            Id = entity.Id;
            Tenant = entity.Tenant;
            FileName = entity.FileName;
            CreateDate = entity.CreateDate;
            Extension = entity.Extension;
            FileSize = entity.FileSize;
            HasFile = entity.HasFile;
            Folder = entity.Folder;
            SmallDocumentId = entity.SmallDocumentId;
            CalculatedFileName = entity.CalculatedFileName;
            IsEncrypted = entity.IsEncrypted;
            MarkForDelete = entity.MarkForDelete;
            SmallDocument = entity.SmallDocument;
        }
        public string Id { get; set; }
        public int Tenant { get; set; }
        public string FileName { get; set; }
        public DateTime? CreateDate { get; set; }
        public string Extension { get; set; }
        public double? FileSize { get; set; }
        public bool HasFile { get; set; }
        public string Folder { get; set; }
        public string SmallDocumentId { get; set; }
        public string CalculatedFileName { get; set; }
        public bool IsEncrypted { get; set; }
        public bool? MarkForDelete { get; set; }
        [ForeignKey("SmallDocumentId")]
        public SmallDocument SmallDocument { get; set; }
    }
}
